app/javascript/components/footprints.vue

```diff
@@ -0,0 +1,111 @@
+<template lang="pug">
+.user-icons(v-if='!loaded')
+  .fa-solid.fa-spinner.fa-pulse
+  |
+  | ロード中
+.user-icons(v-else)
+  h3.user-icons__title
+    | 見たよ
+  ul.user-icons__items(
+    v-if='tenOrLessFootprints || (moreThanTenFoorptints && isDisplay)'
+  )
+    footprint(
+      v-for='footprint in limitedDisplayOnPage',
+      :key='footprint.key',
+      :footprint='footprint'
+    )
+  .user-icons__more(
+    v-if='moreThanTenFoorptints && isDisplay',
+    @click='showRemainingFootprints'
+  )
+    | その他{{ numberOfRemainingFootprints }}人
+  ul.user-icons__items(v-else-if='moreThanTenFoorptints && !isDisplay')
+    footprint(
+      v-for='footprint in footprints',
+      :key='footprint.key',
+      :footprint='footprint'
+    )
+</template>
+<script>
+import Footprint from './footprint'
+
+export default {
+  name: 'Footprints',
+  components: {
+    footprint: Footprint
+  },
+  props: {
+    footprintableId: { type: String, required: true },
+    footprintableType: { type: String, required: true }
+  },
+  data() {
+    return {
+      footprints: [],
+      isDisplay: true,
+      footprintTotalCount: null,
+      loaded: null
+    }
+  },
+  computed: {
+    url() {
+      const params = new URL(location.href).searchParams
+      params.set('footprintableType', this.footprintableType)
+      params.set('footprintableId', this.footprintableId)
+      return `/api/footprints.json?footprintable_type=${this.footprintableType}&footprintable_id=${this.footprintableId}`
+    },
+    limitedDisplayOnPage() {
+      return this.footprints.slice(0, 10)
+    },
+    tenOrLessFootprints() {
+      return this.footprintTotalCount <= 10
+    },
+    moreThanTenFoorptints() {
+      return this.footprintTotalCount > 10
+    },
+    numberOfRemainingFootprints() {
+      return this.footprintTotalCount - 10
+    }
+  },
+  created() {
+    window.onpopstate = function () {
+      location.replace(location.href)
+    }
+    this.getFootprints()
+  },
+  methods: {
+    token() {
+      const meta = document.querySelector('meta[name="csrf-token"]')
+      return meta ? meta.getAttribute('content') : ''
+    },
+    getFootprints() {
+      fetch(this.url, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json; charset=utf-8',
+          'X-Requested-With': 'XMLHttpRequest',
+          'X-CSRF-Token': this.token()
+        },
+        credentials: 'same-origin',
+        redirect: 'manual'
+      })
+        .then((response) => {
+          return response.json()
+        })
```

Comment:
参考情報程度ですが、ここはアロー関数を使っているのでreturnを省略して↓のような書き方ができます。（更に引数のカッコも省略可）
レスポンスを加工するお決まりの処理ということで、割とよく見かける書き方で、bootcampアプリ内でも少数ですがこの書き方の部分もあるようでした。

[アロー関数式 - JavaScript | MDN](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Functions/Arrow_functions#%E5%9F%BA%E6%9C%AC%E7%9A%84%E3%81%AA%E6%A7%8B%E6%96%87)
```suggestion
        .then((response) => response.json())
```

Reply:
アロー関数を使うとreturnを省略できるのですね、知りませんでした！勉強になります🙏

Reply:
@Nabegon アロー関数を使った場合の中で、１行で書く場合だけreturnを省略できます〜